### ISSUE TITLE
Settings page shows "upgrade" button for members of team of billing account with gift subscription 

### ISSUE THREAD

**Issue by catdevnull** [buttondown/roadmap#3222]
![image](https://github.com/user-attachments/assets/83ce685c-fe7d-4c9a-8e78-fa544f05d87d)

confusingly shows up as features that requires an upgrade. Should show the proper feature status instead (left: team member account. right: newsletter owner account)

only happens in gift subscription accounts

**Comment by catdevnull** [comment]
This seems harder than what I can handle at the moment

**Comment by jmduke** [comment]
Yeah, the logic here is a little convoluted (check out DECISION-5 in Notion for more) at first glance. I think all we need to do is also serialize and pull in `owning_account_billing_type` like we do for `owning_account_billing_plan`, but I don't think this is super urgent.

**Comment by catdevnull** [comment]
Okay, do assign this to me whenever you think it should be done then. I just didn't want to touch too much billing code that I didn't know much about, but your solution seems easy and makes sense
